### PR TITLE
Refactor leader selection test

### PR DESCRIPTION
**Pull request by pdex**
Big push to refactor testing.
* Utility class for creating several instances of ChicagoServers with unique config values.
* Chicago has been copied/refactored into ChicagoServer
* Servers store their host:ip in zookeeper instead of host
* Startup and Shutdown should function sanely

### CODE REVIEW

File: src/main/java/com/xjeffrose/chicago/client/ConnectionPoolManager.java [hunk 1]
```diff
@@ -49,14 +54,36 @@ public ConnectionPoolManager(String hostname) {
...
   }
 
+  public void start() {
+    running.set(true);
+  }
+
+  public void stop() {
+    log.info("ConnectionPoolManager stopping");
+    running.set(false);
+    ChannelGroup channelGroup = new DefaultChannelGroup(workerGroup.next());
```
Comment (siddharth20729) [inline src/main/java/com/xjeffrose/chicago/client/ConnectionPoolManager.java]: Where is this defined **workerGroup**? 
Compilation failure:
[ERROR] 
/src/main/java/com/xjeffrose/chicago/client/ConnectionPoolManager.java:[64,57] cannot find symbol

File: src/main/java/com/xjeffrose/chicago/server/ChicagoServer.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      curator.start();
+      curator.blockUntilConnected();
+
+      configureLeaderSelector();
+
+      configureZookeeper();
+
+      dbManager = new DBManager(config);
+      nodeWatcher = new NodeWatcher();
+      nodeWatcher.refresh(zkClient, leaderSelector, dbManager, config);
```
Comment (siddharth20729) [inline src/main/java/com/xjeffrose/chicago/server/ChicagoServer.java]: Please check this as well 
src/main/java/com/xjeffrose/chicago/server/ChicagoServer.java:[71,18] method refresh in class com.xjeffrose.chicago.NodeWatcher cannot be applied to given types;

File: src/main/java/com/xjeffrose/chicago/client/ConnectionPoolManager.java (hunk 1 above)
Comment (pdex) [inline src/main/java/com/xjeffrose/chicago/client/ConnectionPoolManager.java]: fixed

File: src/main/java/com/xjeffrose/chicago/server/ChicagoServer.java (hunk 2 above)
Comment (pdex) [inline src/main/java/com/xjeffrose/chicago/server/ChicagoServer.java]: fixed

**Pull request merged**